pkgs/by-name/xm/xmoji/package.nix

```
@@ -0,0 +1,61 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pkg-config,
+  fontconfig,
+  harfbuzz,
+  libpng,
+  xorg,
+  libxkbcommon,
+  xcb-util-cursor,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  name = "xmoji";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "Zirias";
+    repo = "xmoji";
+    rev =
+      let
+        inherit (lib.versions) majorMinor patch;
+        inherit (finalAttrs) version;
+      in
+      "refs/tags/v${majorMinor version}-${patch version}";
+    hash = "sha256-ZZ1jW97JUv003bAMZZfGWbAAPgeZlpBKREaedFi3R8M=";
+    fetchSubmodules = true;
+  };
+
+  postPatch = ''
+    substituteInPlace zimk/lib/platform.mk \
+      --replace-fail 'PATH:=''$(POSIXPATH)' "#"
+  '';
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    fontconfig
+    harfbuzz
+    libpng
+    libxkbcommon
+    xcb-util-cursor
+  ] ++ (with xorg; [
+    xcbutil
+    libXcursor
+    xcbutilimage
+  ]);
```

```suggestion
    xcbutil
    libXcursor
    xcbutilimage
  ]
```
attrs under `xorg` can be passed in directly